### DISCUSSION TITLE
Friendly way to upgrade Elementary

### DISCUSSION THREAD

**Discussion by crisconru** [elementary/triage#346] [Ideas]
First of all, I don't understand deeply the implications of my proposal, so maybe I should be quiet. 

I am a Linux user for many years so I'm used to format my computer with a new version of Elementary. But I think for newbies user, and this distro put a ton of efforts in the users, it is not the best approach.

I would suggest to move to a strategy like Manjaro, based on a rolling release distro, and block the upgrade until the proper moment to do that. Maybe this is a crazy or risky solution and it was discarded in the past.

Maybe it's possible to create a tool to upgrade the distro. I remember when I started using Ubuntu (9 or 11 version), there was a way to upgrade between versions. But after that, the installation was slower than if i installed from scratch.

I don't know what it's the solution. But I think the current format is a pain in the ass at the end of the day for all of us and maybe for you (the developers) it's a friction point too to work with completely differents versions instead of working in a iterational and incremental version (rolling release).

**Comment by wasantosfi** [comment]
I agree with you.
Most distros we can upgrade it simply by using dist-upgrade command, which is highly appreciated if we don't have much time to format, install our needed softwares and setting all over again.
Your idea should be prioritized.

**Comment by danirabbit** [comment]
Just to recap the state of things, we have an OS upgrades project board here: https://github.com/orgs/elementary/projects/103

We've been roadblocked a few times by missing features in PackageKit. In general there's a number of problems that have caused failures in the upgrade process in our prototypes, including things like configuration file changes. We tend to have fairly large breaking changes under the hood between major releases and so it's pretty challenging to reliably upgrade a system across that gap.

Something we've been looking into more lately is something like OSTree where we would have image-based updates. See this branch for progress there: https://github.com/elementary/os/pull/582

This is definitely both a very important and very challenging problem to solve!

**Comment by wasantosfi** [comment]
Hi @danirabbit , looks like the last update of the referred branch was months ago (end of February).
But I think this feature would be really welcome not only for us end users, but to adquire new users and sponsors, getting more visibility between concurrents as well.
So, I think (as an external agent) it is a good idea to be prioritized both on usability and on business purposes.

**Comment by edmundlaugasson** [comment]
As Elementary OS is based on Ubuntu LTS, like Linux Mint, you can consider to use its [mintupgrade](https://github.com/linuxmint/mintupgrade).
